### PR TITLE
feat: add option to keep source maps

### LINKED ISSUE #215: Sourcemaps are not emitted when excluded with `deleteOriginalAssets` option
- Operating System: MacOs Catalina 
- Node Version: 14.15
- NPM Version: 6.14.8
- webpack Version: 5.4.0
- compression-webpack-plugin Version: 6.0.5

### Expected Behavior

Sourcemap should be emitted with `deleteOriginalAssets` when excluded from compression.

### Actual Behavior

When `deleteOriginalAssets` is enabled and map files are excluded from the compression. It is expected that `js` files are compressed with the same filename and map files are emitted as it is. but the map files are not emitted.

### Code

```js
// webpack.config.js
const CompressionPlugin = require('compression-webpack-plugin')
const FileManagerPlugin = require('filemanager-webpack-plugin')

module.exports = {
  mode: 'production',
  devtool: 'source-map',
  plugins: [
    new FileManagerPlugin({
      events: {
        onStart: {
          delete: ['dist/']
        }
      }
    }),
    new CompressionPlugin({
      filename: '[file]',
      exclude: /.map$/,
      minRatio: 10,
      deleteOriginalAssets: true
    })
  ]
}

```

### How Do We Reproduce?

Run webapck with the given config. the dist folder should have a compressed `main.js` and uncompressed `main.js.map`, but the map file is not emitted.

### PR DESCRIPTION
**Pull request by sibiraj-s**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Fixes #215 . Added option to not delete sourcemap when deleting the originalAsset

### Breaking Changes

none

### Additional Info

none

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -289,7 +289,7 @@ module.exports = {
 
 ### `deleteOriginalAssets`
 
-Type: `Boolean`
+Type: `Boolean|keep-source-map`
```
Comment (evilebottnawi) [inline README.md:292]: `Boolean | 'keep-source-map'`

File: src/index.js [hunk 2]
```diff
@@ -299,7 +299,21 @@ class CompressionPlugin {
           }
 
           if (this.options.deleteOriginalAssets) {
-            // eslint-disable-next-line no-param-reassign
+            if (this.options.deleteOriginalAssets === 'keep-source-map') {
+              // TODO `...` required only for webpack@4
+              const updatedAssetInfo = {
+                ...info,
+                related: { ...info.related, sourceMap: null },
+              };
```
Comment (evilebottnawi) [inline src/index.js:307]: Let's do it better:
```
delete updatedAssetInfo.related.sourceMap;
```

Comment (sibiraj-s) [reply src/index.js:307]: I tried this, this is not working. The sourceMap is getting added back if a value is not provided.

File: test/__snapshots__/deleteOriginalAssets.test.js.snap.webpack5 [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+  Array [
+    "09a1a1112c577c2794359715edfcb5ac.png",
+    78117,
+  ],
+  Array [
+    "23fc1d3ac606d117e05a140e0de79806.svg",
+    393,
+  ],
+  Array [
+    "async.async.1c2911c7c819fc826fb7.js",
```
Comment (evilebottnawi) [inline test/__snapshots__/deleteOriginalAssets.test.js.snap.webpack5:14]: hm, I think we should have source map here

Comment (sibiraj-s) [reply test/__snapshots__/deleteOriginalAssets.test.js.snap.webpack5:14]: Source map was not generated for that.

```js
  [
      '09a1a1112c577c2794359715edfcb5ac.png',
      '23fc1d3ac606d117e05a140e0de79806.svg',
      'main.6a5afd3ff5b2f0fa8aaa.js',
      'async.async.1c2911c7c819fc826fb7.js',
      'main.6a5afd3ff5b2f0fa8aaa.js.map'
    ]
```

these are the total assets included for compression.

**Comment by sibiraj-s** [comment]
I am not sure why tests are failing. I tested with both webpack 4 and 5 in local everything works fine 🤔

**Comment by sibiraj-s** [comment]
Ah, Found it. latest version `webpack is v5.4` in CI, in `local it is v5.3`. I have update the dep to 5.4. let me know. I need to revert it or raise it in a separate PR.

File: src/index.js (hunk 2 above)
Comment (sibiraj-s) [reply src/index.js:307]: Webpack I think is doing a smart merge there. 🤔

**Comment by evilebottnawi** [comment]
All is fine

**Review by evilebottnawi — APPROVED** [review]

**Pull request merged**